pyproject.toml

```
@@ -0,0 +1,109 @@
+[build-system]
+requires = ["setuptools>=61.0", "setuptools_scm[toml]>=6.2"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools_scm]
+
+[project]
+name = "libpysal"
+dynamic = ["version"]
+authors = [ # in alphabetical order
+    { name = "Martin Fleischmann", email = "martin@martinfleischmann.net" },
+    { name = "Eli Knaap", email = "ek@knaaptime.com" },
+    { name = "Serge Rey", email = "sjsrey@gmail.com" },
+    { name = "Levi Wolf", email = "levi.john.wolf@gmail.com" },
+]
+maintainers = [{ name = "PySAL Developers" }]
+license = { text = "BSD 3-Clause" }
+description = "Core components of PySAL A library of spatial analysis functions"
+keywords = ["spatial statistics", "spatial graphs"]
+readme = "README.md"
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Intended Audience :: Science/Research",
+    "Topic :: Scientific/Engineering :: GIS",
+]
+requires-python = ">=3.7"
```

Minimum is now [3.8](https://github.com/pysal/libpysal/blob/357669184c35f6d8ce86d01e9412bb24b09b8353/.github/workflows/unittests.yml#L26), so we can bump this.

```suggestion
requires-python = ">=3.8"
```